### DISCUSSION TITLE
DataValidationErrors not handling asynchronous validation

### DISCUSSION THREAD

**Discussion by mrj001** [AvaloniaUI/Avalonia#15377] [Q&A]
### Describe the bug

Using INotifyDataErrorInfo, when the ErrorsChanged event is fired (from a background validation task), the DataValidationErrors is not updated.

### To Reproduce

I prepared a minimal reproduction of this in a 7kb .tgz file.  However, when I drop it here, I get the following error.  Note that "TGZ" is listed as an acceptable file type.  The same error happened with the uncompressed tar file.  I tried the compressed version with both lower case and upper case "TGZ" extension.

![Screenshot_20240411_133157](https://github.com/AvaloniaUI/Avalonia/assets/62225294/edc7ffe7-4048-430f-bae9-bb351aa1f836)

In this repro, I used a Task (with a 2.5 second delay) to validate the data in a TextBox.  Upon entering invalid data (any text containing 'q'), the ErrorsChanged event is fired, but the DataValidationErrors control does not call GetErrors, and does not update to show the validation error.  Upon the next keystroke (which does not correct the error), the error is then displayed.

### Expected behavior

I expected that when the ErrorsChanged event is fired, the DataValidationErrors control would update to show the error.

### Avalonia version

11.0.10

### OS

Linux

**Comment by maxkatz6** [comment]
I definitely did use async validation via INotifyDataErrorInfo back in 0.10 days. It might have been broken since then, but we need a minimal repro at the very least. It could be a problem in the user code as well.

**Comment by timunie** [comment]
> I expected that when the ErrorsChanged event is fired, the DataValidationErrors control would update to show the error.

I believe the Event needs to be sent from UiThread using `Dispatcher`

**Comment by mrj001** [comment]
Aha!  I tried uploading the repro in Safari on MacOS, and it worked.

[repro.tgz](https://github.com/AvaloniaUI/Avalonia/files/14959521/repro.tgz)

@timunie, the event is sent to the UI thread.

**Comment by maxkatz6** [comment]
@mrj001 thanks. There is a typo in your code:
```
               _errorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(nameof(propName))));
```
should be
```
               _errorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(propName)));
```
